### DISCUSSION TITLE
Support multiple `ApolloClient` instances

### DISCUSSION THREAD

**Discussion by mfulton26** [apollographql/apollo-client-devtools#920] [Ideas]
My app has more than one `ApolloClient` instance. I would like to be able to select which client to use when making queries, etc.

**Comment by ajzozakiewicz** [comment]
We would also like to see this functionality added. A drop down to toggle between apollo client instances would be very nice!

**Reply by jerelmiller** [reply]
Hey @ajzozakiewicz 👋 

I didn't realize this discussion was open, but this feature was added to devtools via https://github.com/apollographql/apollo-client-devtools/pull/1418 back with version [4.17.0](https://github.com/apollographql/apollo-client-devtools/releases/tag/v4.17.0) which released in July 2024.

Apollo Client [3.11](https://github.com/apollographql/apollo-client/releases/tag/v3.11.0) added support for naming clients so they are identifiable in devtools. As long as you're updated to that version, you can use the new `devtools` option to enable the client to connect to devtools and name the client.

```ts
new ApolloClient({
  devtools: {
    enabled: true,
    name: "My Custom Client",
  },
});
```

Please make sure to update devtools to at least 4.17.0 and you should have this feature!

**Reply by ajzozakiewicz** [reply]
Oh awesome. Thank you!

**Comment by jerelmiller** [comment]
This feature was completed in [4.17.0](https://github.com/apollographql/apollo-client-devtools/releases/tag/v4.17.0) and released in July 2024. As such I'm closing this discussion as completed.
